### PR TITLE
Cloud compactor schedule

### PR DESCRIPTION
**Pull request by lightningrob**
Added command line tool as well.

### CODE REVIEW

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  @Override
+  public void run() {
+    Timer.Context docTimer = vcrMetrics.blobCompactionTime.time();
+    try {
+      compactPartitions();
+    } finally {
+      docTimer.stop();
+    }
```
Comment (cgtz) [inline ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java]: One gotcha i've found with ScheduledExecutorService is that, if an execution throws an exception, future executions of the runnable will stop: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleAtFixedRate(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)

I would recommend catching any runtimeexceptions (Maybe throwables too?) and logging here.

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      VcrMetrics vcrMetrics, boolean testMode) {
+    this.cloudDestination = cloudDestination;
+    this.partitionMap = partitionMap;
+    this.vcrMetrics = vcrMetrics;
+    this.testMode = testMode;
+  }
+
+  @Override
+  public void run() {
+    Timer.Context docTimer = vcrMetrics.blobCompactionTime.time();
```
[the remark this answers is not in the document]
Comment (lightningrob) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java]: Bad copy paste.  Will fix.

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  private final boolean testMode;
+
+  /**
+   * Public constructor.
+   * @param cloudDestination the cloud destination to use.
+   * @param partitionMap the map whose keys are the partitions to compact.
+   * @param vcrMetrics the metrics to update.
+   * @param testMode if true, the compaction methods only perform the query but do not actually remove the blobs.
+   */
+  public CloudStorageCompactor(CloudDestination cloudDestination, Map<PartitionId, PartitionInfo> partitionMap,
```
[the remark this answers is not in the document]
Comment (lightningrob) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java]: True.  For some reason I was nervous about passing the keyset of a mutable map, even though it's backed by the map and should be safe.

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java (hunk 1 above)
Comment (lightningrob) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java]: Will do.

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lightningrob) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java]: Changed to set.

**Pull request merged**